### DISCUSSION TITLE
Rocket thumbnail generation

### DISCUSSION THREAD

**Discussion by jadebenn** [DarkflameUniverse/DarkflameServer#1264] [General]
I remember that on live you'd eventually get a little thumbnail of your rockets, allowing you to tell which was which. Is there any way to do that on the Darkflame client currently? If not, what is known about that system and how it worked?

**Comment by EmosewaMC** [comment]
There is no way to do that easily right now.  Only way to do it right now is to generate them manually, make a code modification to give rockets and cars subkeys, and then name the generated dds file with that subkey and place that file in the `UserMade/<ugcId>`.

**Reply by jadebenn** [reply]
Right, I'm not asking for an implementation, just trying to understand the technical barriers. How was this handled on Live? Was the completed model sent off to a server to generate the .dds and then sent back to the client?

**Reply by EmosewaMC** [reply]
How live did it is your rocket/car would get sent to a headless renderer to generate the image and when it was done your client would download it from the remote.  The Image you see right now is just a temp one since asking a 2011 computer to do this is very expensive, so they didn't bother letting it.

**Reply by jadebenn** [reply]
I'm guessing a similar system was used by brick by brick models in live to apply the rendering effects as well. I remember how it always took a bit for the "flat" models to get the fancy rendering.

**Reply by EmosewaMC** [reply]
Yes that would be the client generating its nif, hkx and lxfml files and then later on, the server would deliver a highly optimized nif, hkx and lxfml for the client to use instead.

**Comment by MustacheMan2** [comment]
> How live did it is your rocket/car would get sent to a headless renderer to generate the image and when it was done your client would download it from the remote. The Image you see right now is just a temp one since asking a 2011 computer to do this is very expensive, so they didn't bother letting it.

Odd to think that they figured making icons was too taxing but generating potentially very large brick models, something in its early stage at the time, was a-ok lol. Anyways, can the server currently even make the distinction on which rocket/car is which? I recall it just picking the first one built regardless of which one is selected.

**Reply by aronwk-aaron** [reply]
the same thing would happen with models as well. DLU just tricks the client into doing it lazily for the sake of it being "functional"

And yes, it can for rockets, but not cars, that has to be implemented

**Reply by EmosewaMC** [reply]
~~This already is implemented in #1266.  It is now in the users' hands to generate these images and distribute them~~

I was thinking of something different unrelated to the ugc thing.  The cars & rockets do now have subkeys, so you can tell them apart like that, so this actually is an easy implementation now.

**Reply by MustacheMan2** [reply]
I am now more confused than before.

**Reply by EmosewaMC** [reply]
Rockets are always able to be told apart based on last used or what you drag.
Cars are first in inventory for some reason
